pkg/kds/client/sink.go

```diff
@@ -47,10 +47,12 @@ func (s *kdsSink) Start(stop <-chan struct{}) (errs error) {
 
 		clusterID, rs, err := s.kdsStream.Receive()
 		if err != nil {
+			if err == io.EOF {
+				break
```

Comment:
Why do we `break` on EOF? I guess `s.kdsStream.Receive()` on the next iteration will return EOF as well

Reply:
There won't be the next iteration without this if, it will use
```
return errors.Wrap(err, "failed to receive a discovery response")
```